### PR TITLE
Feature/infospecies interfaces

### PR DESCRIPTION
**Pull request by cloedu87**
- [x] generate dwc archive
- [x] send mail to inform infospecies about new content to check
- [x] buid API endpoint to receive a dwc-archive to import and update the approval layer with new incomming data
- [x] tests
- [x] docs
- [x] little fixes and glamour
- [x] changes among the import and encoding process to fit approval needs
- [x] resources and migrations

fixes the following issues:
https://infofauna-support.atlassian.net/browse/SCN-210
https://infofauna-support.atlassian.net/browse/SCN-189
https://infofauna-support.atlassian.net/browse/SCN-266
https://infofauna-support.atlassian.net/browse/SCN-267

### CODE REVIEW

File: lib/data_aggregator/records/approval/helpers.ex [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+      end
+    end
+
+    if Records.execute_async?() do
+      upload_fn
+      |> Task.async()
+      |> Task.await()
+    else
+      upload_fn.()
+    end
```
Comment (hwuethrich) [inline lib/data_aggregator/records/approval/helpers.ex:207]: Why `Task.async()` if we await the result?

File: lib/data_aggregator/records/publication/info_species.ex [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  defp update_records_approval_started_at(query) do
+    query
+    |> Records.stream!(page: false)
+    |> Stream.map(&process_record(&1))
+    |> Stream.run()
+  end
+
+  defp process_record(record) do
+    Record.update_last_approval_started_at!(record)
+  end
```
Comment (hwuethrich) [inline lib/data_aggregator/records/publication/info_species.ex:110]: Could probably be optimised into an atomic update (single SQL `UPDATE ... WHERE ...`)

**Review by hwuethrich — APPROVED** [review]
Can't verify the functionality, but LGTM! :)

File: lib/data_aggregator/records/approval/helpers.ex (hunk 1 above)
Comment (mbaertschi) [reply lib/data_aggregator/records/approval/helpers.ex:207]: I know, the wording is wierd, but this is the way you should process concurrent tasks and await them in elixir...
https://hexdocs.pm/elixir/1.16.3/Task.html#module-async-and-await

**Pull request merged**